README.md

```diff
@@ -11,78 +11,114 @@ related-issues: (will contain links to implementation PRs)
 # Summary
 [summary]: #summary
 
-Make most attribute definitions in `pkgs/top-level/all-packages.nix` be auto-generated from a single flat directory instead, where the subdirectory corresponds to the attribute name. The ad-hoc category-based structure of packages will be gotten rid of.
+Make trivial top-level attribute definitions in `pkgs/top-level/all-packages.nix` be auto-generated from a predictable attribute-based file hierarchy.
```

Comment:
```suggestion
Auto-generate trivial top-level attribute definitions in `pkgs/top-level/all-packages.nix`  from a sharded directory that matches the attribute name.
```

 - Simplify sentence structure.
 - Predictable hierarchy suggests actual hierarchy. I don't like adding sharded here, but just _directory_ would be equally wrong, compared to _hierarchy_.